rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrades.swift

```diff
@@ -54,24 +97,254 @@ struct SubmissionGrades: View {
+                saveGrade(excused: true)
+            } else if assignment.gradingType == .percent {
+                saveGrade("\(round(value / max(assignment.pointsPossible ?? 0, 0.01) * 100))%")
+            } else if assignment.gradingType == .points {
+                saveGrade(String(round(value)))
+            }
+        }
+    }
+
+    func sliderChangedValue(_ value: Double) {
+        sliderValue = value
+        sliderTimer?.invalidate()
+        sliderTimer = nil
+        if value == 0 {
+            sliderTimer = Timer.scheduledTimer(withTimeInterval: 1, repeats: false) { _ in
+                sliderCleared = true
+                sliderExcused = false
+                UISelectionFeedbackGenerator().selectionChanged()
+            }
+        } else if value == assignment.pointsPossible {
+            sliderTimer = Timer.scheduledTimer(withTimeInterval: 1, repeats: false) { _ in
+                sliderCleared = false
+                sliderExcused = true
+                UISelectionFeedbackGenerator().selectionChanged()
+            }
+        } else {
+            sliderCleared = false
+            sliderExcused = false
+        }
+    }
+
+    // MARK: Prompt for an updated grade
+
+    func promptNewGrade() {
+        var message: String?
+        switch assignment.gradingType {
+        case .gpa_scale:
+            message = NSLocalizedString("GPA")
+        case .letter_grade:
+            message = NSLocalizedString("Letter grade")
+        case .not_graded, .pass_fail:
+            message = nil
+        case .percent:
+            message = NSLocalizedString("Percent (%)")
+        case .points:
+            message = assignment.outOfText
+        }
+        let prompt = UIAlertController(title: NSLocalizedString("Customize Grade"), message: message, preferredStyle: .alert)
+        if assignment.gradingType == .pass_fail {
+            prompt.addAction(AlertAction(NSLocalizedString("Complete")) { _ in saveGrade("complete") })
+            prompt.addAction(AlertAction(NSLocalizedString("Incomplete")) { _ in saveGrade("incomplete") })
         } else {
-            Icon.addSolid.foregroundColor(Color(Brand.shared.linkColor))
+            prompt.addTextField { field in
+                field.placeholder = NSLocalizedString("")
+                field.returnKeyType = .done
+                field.text = submission.excused == true ? NSLocalizedString("Excused") :
+                    hasLateDeduction ? submission.enteredGrade : submission.grade
+                field.addTarget(prompt, action: #selector(UIAlertController.performOKAlertAction), for: .editingDidEndOnExit)
+            }
+        }
+        prompt.addAction(AlertAction(NSLocalizedString("No Grade")) { _ in saveGrade("") })
+        if submission.excused != true {
+            prompt.addAction(AlertAction(NSLocalizedString("Excuse Student")) { _ in saveGrade(excused: true) })
+        }
+        if assignment.gradingType != .pass_fail {
```

Comment:
Do we need to check for `.not_graded` here?

Reply:
The button with the action to get here isn't shown for `.not_graded`.